diff --git a/include/crow/http_request.h b/include/crow/http_request.h
index 1111111..2222222 100644
--- a/include/crow/http_request.h
+++ b/include/crow/http_request.h
@@ -43,9 +43,9 @@ namespace crow
         std::string body;
         std::string remote_ip_address; ///< The IP address from which the request was sent.
         unsigned char http_ver_major, http_ver_minor;
-        bool keep_alive,    ///< Whether or not the server should send a `connection: Keep-Alive` header to the client.
-          close_connection, ///< Whether or not the server should shut down the TCP connection once a response is sent.
-          upgrade;          ///< Whether or noth the server should change the HTTP connection to a different connection.
+        bool keep_alive = false,    ///< Whether or not the server should send a `connection: Keep-Alive` header to the client.
+          close_connection = false, ///< Whether or not the server should shut down the TCP connection once a response is sent.
+          upgrade = false;          ///< Whether or noth the server should change the HTTP connection to a different connection.

         void* middleware_context{};
         void* middleware_container{};
